bin/lint-i18n-translations.js

```diff
@@ -40,11 +54,146 @@ async function loadTranslations (langDir) {
     return translations
 }
 
+async function isAnyMessagesDirectoryExists (langDir) {
+    try {
+        const locales = await fs.readdir(langDir, { withFileTypes: true })
+        for (const locale of locales) {
+            if (!locale.isDirectory()) continue
+            // nosemgrep: javascript.lang.security.audit.path-traversal.path-join-resolve-traversal.path-join-resolve-traversal
+            const localePath = path.join(langDir, locale.name)
+            // nosemgrep: javascript.lang.security.audit.path-traversal.path-join-resolve-traversal.path-join-resolve-traversal
+            const messagesPath = path.join(localePath, 'messages')
+            try {
+                const stat = await fs.stat(messagesPath)
+                if (stat.isDirectory()) {
+                    return true
+                }
+            } catch (e) {
+                // Ignore errors that indicate messages directory does not exist
+                if (e.code !== 'ENOENT') {
+                    throw new Error(`Error checking messages directory at "${messagesPath}": ${e.message}`)
+                }
+            }
+        }
+    } catch (e) {
+        throw new Error(`Error reading lang directory: ${e.message}`)
+    }
+    return false
+}
+
+async function loadMessagesDirectories (langDir) {
+    const messageData = {}
+    try {
+        const locales = await fs.readdir(langDir, { withFileTypes: true })
+        for (const locale of locales) {
+            if (!locale.isDirectory()) continue
+            // nosemgrep: javascript.lang.security.audit.path-traversal.path-join-resolve-traversal.path-join-resolve-traversal
+            const localePath = path.join(langDir, locale.name, 'messages')
+            // nosemgrep: javascript.lang.security.audit.path-traversal.path-join-resolve-traversal.path-join-resolve-traversal
+            console.log(`Loading files ${path.join(localePath, '*')}`)
+            messageData[locale.name] = {}
+            try {
+                const entries = await fs.readdir(localePath, { withFileTypes: true })
+                for (const entry of entries) {
+                    // nosemgrep: javascript.lang.security.audit.path-traversal.path-join-resolve-traversal.path-join-resolve-traversal
+                    const messageDirPath = path.join(localePath, entry.name)
+                    if (entry.isDirectory()) {
+                        try {
+                            const files = await fs.readdir(messageDirPath)
+                            const njkFiles = {}
+                            for (const file of files) {
+                                // nosemgrep: javascript.lang.security.audit.path-traversal.path-join-resolve-traversal.path-join-resolve-traversal
+                                const filePath = path.join(messageDirPath, file)
+                                if (file.endsWith('.njk')) {
+                                    try {
+                                        await fs.access(filePath)
```

Comment:
How about to nest code even more and wrap every single line in try-catch? Because right now the code reads too easily :)

Reply:
I guess we need to refactor this part heavily, right now its heavy 400-liner, no one can read and understand.
I hate this kind of files, because you cannot even tell somebody to fix something in it.

It's like `@open-condo/next`, very heavy and hard to understand files with unstructured code...

Reply:
You can partially fix it with naming (for example this utils scans for push / email templates, you can call it `loadNotificationTemplates` or something, remove nested try catches here, panic is enough for bin